testsupport/wait/awaitility.go

```diff
@@ -892,3 +892,52 @@ func For[T client.Object](t *testing.T, a *Awaitility, obj T) *Waiter[T] {
 		gvk:   gvks[0],
 	}
 }
+
+// doUpdate tries to update the Status or Spec of the given object
+// If it fails with an error (for example if the object has been modified) then it retrieves the latest version and tries again
+// Returns the updated object
+func (w *Waiter[T]) doUpdate(status bool, objectName string, modify func(T)) (T, error) {
+	var objectToReturn T
+	err := wait.PollUntilContextTimeout(context.TODO(), w.await.RetryInterval, w.await.Timeout, true, func(ctx context.Context) (done bool, err error) {
+		obj := &unstructured.Unstructured{}
+		obj.SetGroupVersionKind(w.gvk)
+		if err := w.await.Client.Get(context.TODO(), types.NamespacedName{Namespace: w.await.Namespace, Name: objectName}, obj); err != nil {
+			return true, err
+		}
+		object, err := w.cast(obj)
+		if err != nil {
+			return false, fmt.Errorf("failed to cast the object to GVK %v: %w", w.gvk, err)
+		}
+		modify(object)
+		if status {
+			// Update the Status
+			if err := w.await.Client.Status().Update(context.TODO(), object); err != nil {
+				w.t.Logf("error updating '%v' Status '%s': %s. Will retry again...", w.gvk, objectName, err.Error())
+				return false, err
+			}
+		} else {
+			// Update the Spec
+			if err := w.await.Client.Update(context.TODO(), object); err != nil {
+				w.t.Logf("Error updating '%v' Spec '%s': %s. Will retry again...", w.gvk, objectName, err.Error())
+				return false, err
```

Comment:
I just noticed that I made a mistake in my suggestion, we should not return an error here; otherwise, the poll loop immediately exists with an error:
```suggestion
				return false, nil
			}
		} else {
			// Update the Spec
			if err := w.await.Client.Update(context.TODO(), object); err != nil {
				w.t.Logf("Error updating '%v' Spec '%s': %s. Will retry again...", w.gvk, objectName, err.Error())
				return false, nil
```

Reply:
is this true also for the status update faliure : https://github.com/codeready-toolchain/toolchain-e2e/pull/1074/files#diff-4abc90985c7fe1e595e0d6f5131e1f4757a6f00c9239f40ae09375d12f15e530R916

should we not return an error here as well and retry ?

Reply:
Yes, it is addressed in line 916 of @MatousJobanek  suggestion! Goinggggg!!

Locally, it passed for me. That's why I did not notice it. Sorry!

Reply:
oh nice, sorry I've missed that from the suggestion, never mind 👍